File: src/main/java/ru/kaplaan/dto/mapper/NoteMapper.java
package ru.kaplaan.dto.mapper;

import org.mapstruct.Mapper;
import ru.kaplaan.domain.entity.Note;
import ru.kaplaan.web.dto.note.NoteDto;

import java.util.List;

@Mapper(componentModel = "spring")
public interface NoteMapper {
    NoteDto toDto(Note note);

    List<NoteDto> toDto(List<Note> notes);

    Note toEntity(NoteDto noteDto);


}

File: notes-server/src/main/kotlin/ru/kaplaan/notes/web/mapper/NoteMapper.kt
package ru.kaplaan.notes.web.mapper

import ru.kaplaan.notes.domain.entity.Note
import ru.kaplaan.notes.web.dto.NoteDto


fun Note.toDto(): NoteDto {
    return NoteDto(
        title = title,
        text = text
    )
}


fun NoteDto.toEntity(owner: String): Note{
    return Note(
        title = title,
        text = text,
        owner = owner
    )
}


fun List<Note>.toDto(): List<NoteDto>{
    val list = arrayListOf<NoteDto>()

    for(note in this){
        list.add(note.toDto())
    }
    return list
}